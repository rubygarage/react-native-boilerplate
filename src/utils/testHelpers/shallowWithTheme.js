import React from 'react';
import { ThemeProvider } from 'styled-components/native';
// eslint-disable-next-line import/no-extraneous-dependencies
import { shallow } from 'enzyme';

import mockedTheme from './mockedTheme';

export default (tree) => shallow(
  <ThemeProvider theme={mockedTheme}>
    {tree}
  </ThemeProvider>,
);
