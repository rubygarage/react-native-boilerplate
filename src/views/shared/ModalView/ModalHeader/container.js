import React from 'react';
import { withTheme } from 'styled-components/native';

import ModalHeaderComponent from './component';

const ModalHeader = (props) => (
  <ModalHeaderComponent {...props} />
);

export { ModalHeader as ModalHeaderContainer };
export default withTheme(ModalHeader);
