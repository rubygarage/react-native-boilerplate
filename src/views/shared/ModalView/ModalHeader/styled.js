import styled from 'styled-components/native';

const Wrap = styled.View`
  position: relative;
  flex-direction: row;
  align-items: center;
  flex-shrink: 0;
  padding: 16px 24px;
  border-bottom-color: rgba(82,97,115, 0.12);
  border-bottom-width: 1;
  border-style: solid;
`;

const Title = styled.Text`
  font-size: 16;
  line-height: 24;
  letter-spacing: -0.2;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.black};
`;

const Close = styled.TouchableOpacity`
  flex-shrink: 0;
  margin-right: 16;
`;

export { Wrap, Title, Close };
