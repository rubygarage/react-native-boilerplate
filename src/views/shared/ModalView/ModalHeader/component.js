import React from 'react';
import PropTypes from 'prop-types';
import Icon from 'react-native-vector-icons/Entypo';

import * as S from './styled';
import useContainer from './hook';

const ModalHeaderComponent = ({ children, isClosable, onClose }) => {
  const theme = useContainer();

  return (
    <S.Wrap>
      {isClosable
      && (
        <S.Close onPress={onClose}>
          <Icon name="cross" size={24} color={theme.colors.grey500} />
        </S.Close>
      )}
      <S.Title>
        {children}
      </S.Title>
    </S.Wrap>
  );
};

ModalHeaderComponent.propTypes = {
  children: PropTypes.node.isRequired,
  isClosable: PropTypes.bool,
  onClose: PropTypes.func,
};

ModalHeaderComponent.defaultProps = {
  isClosable: false,
  onClose: undefined,
};

export default ModalHeaderComponent;
