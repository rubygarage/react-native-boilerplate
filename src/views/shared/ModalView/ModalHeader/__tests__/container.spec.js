import React from 'react';
import { View } from 'react-native';

import shallowWithTheme from 'utils/testHelpers/shallowWithTheme';
import diveTo from 'utils/testHelpers/diveToEnzyme';
import ModalHeader, { ModalHeaderContainer } from '../container';


describe('ModalHeader container', () => {
  const wrapper = shallowWithTheme(<ModalHeader><View /></ModalHeader>);
  const container = diveTo(wrapper, ModalHeaderContainer);

  it('renders correctly default', () => {
    expect(container).toMatchSnapshot();
  });
});
