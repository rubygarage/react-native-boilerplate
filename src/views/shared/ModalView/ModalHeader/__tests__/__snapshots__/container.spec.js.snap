// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalHeader container renders correctly default 1`] = `
<ModalHeaderComponent
  isClosable={false}
  theme={
    Object {
      "colors": null,
      "fonts": null,
      "sizes": null,
      "type": "Mocked theme",
    }
  }
>
  <View />
</ModalHeaderComponent>
`;
